x/evm/keeper/state_transition.go

```diff
@@ -167,6 +167,9 @@ func (k *Keeper) ApplyTransaction(ctx sdk.Context, tx *ethtypes.Transaction) (*t
 	// pass true to commit the StateDB
 	res, err := k.ApplyMessageWithConfig(tmpCtx, msg, nil, true, cfg, txConfig)
 	if err != nil {
+		// when a transaction contains multiple msg, as long as one of the msg fails
+		// all gas will be deducted. so is not msg.Gas()
+		k.ResetGasMeterAndConsumeGas(ctx, ctx.GasMeter().Limit())
```

Comment:
@luchenqun @GAtom22 
There is another `ApplyMessage` method does use `ApplyMessageWithConfig`.
This should be handled there either?

Reply:
Nvm, it has none of business with gas